employees/tests/test_unit_custom_mixins.py

```diff
@@ -0,0 +1,132 @@
+        self
+    ):
+        mixin = MonthNavigationMixin()
+        self.assertEqual(mixin._get_title_date(year=2019, month=4), "04/19")
+
+    def test_get_next_month_url_should_generate_url_for_next_month_for_given_path(self):
+        mixin = MonthNavigationMixin()
+        mixin.url_name = "project-report-list"
+        self.assertEqual(mixin._get_next_month_url(2018, 12, 1), "/reports/project/1/2019/1/")
+
+    def test_get_previous_month_url_should_generate_url_for_previous_month_for_given_path(self):
+        mixin = MonthNavigationMixin()
+        mixin.url_name = "project-report-list"
+        self.assertEqual(mixin._get_previous_month_url(2019, 1, 1), "/reports/project/1/2018/12/")
+
+    def test_get_recent_month_url_should_generate_url_for_current_month_for_given_path(self):
+        current_date = datetime.datetime.now()
+        mixin = MonthNavigationMixin()
+        mixin.url_name = "project-report-list"
+        self.assertEqual(
+            mixin._get_recent_month_url(1), f"/reports/project/1/{current_date.year}/{current_date.month}/"
+        )
+
+
+class MonthNavigationMixinContextDataTests(TestCase):
+    def setUp(self):
+        self.mixin = MonthNavigationMixin()
+        self.mixin.url_name = "project-report-list"
+        self.mixin.__setattr__("kwargs", {})
+
+    def _get_month_navigation_context_data(self, year, month, pk):
+        request = HttpRequest()
+        request.path = f"/reports/project/{pk}/{year}/{month}/"
+        self.mixin.request = request
+        self.mixin.kwargs["year"] = year
+        self.mixin.kwargs["month"] = month
+        self.mixin.kwargs["pk"] = pk
+        return Context(self.mixin._get_month_navigator_params())
+
+    def _render_month_navigation_bar(self, year, month, pk):
+        context = self._get_month_navigation_context_data(year, month, pk)
+        template_to_render = Template("{% include 'employees/month_navigation_bar.html' %}")
+        return template_to_render.render(context)
+
+    def test_month_navigator_get_context_data_method_should_return_all_data_necessary_for_month_navigator_template(
+        self
+    ):
+        context = self._get_month_navigation_context_data(2019, 3, 1)
+        current_date = datetime.datetime.now()
+        expected_output = {
+            "path": self.mixin.request.path,
+            "navigation_text": MonthNavigationText,
+            "month_form": MonthSwitchForm(initial_date=datetime.date(year=2019, month=3, day=1)),
+            "next_month_url": "/reports/project/1/2019/4/",
+            "recent_month_url": f"/reports/project/1/{current_date.year}/{current_date.month}/",
+            "previous_month_url": "/reports/project/1/2019/2/",
+            "disable_next_button": False,
+            "disable_previous_button": False,
+            "title_date": "03/19",
+        }
+        for key in expected_output.keys():  # pylint: disable=consider-iterating-dictionary
+            if key == "month_form":
```

Comment:
`MonthNavigationText` class under presented under **navigation_text** cannot be compared using '__eq__', it causes an error in test (I guess it's because it's not an instance?).
`MonthSwitchForm` is the only object in the whole package that causes an issue on `self.assertEqual()`, therefore there's this differentiation.

Reply:
please see above

Reply:
Done